python/cudf/cudf/_lib/pylibcudf/strings/replace.pyx

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+
+from cudf._lib.pylibcudf.column cimport Column
+from cudf._lib.pylibcudf.libcudf.column.column cimport column
+from cudf._lib.pylibcudf.libcudf.scalar.scalar cimport string_scalar
+from cudf._lib.pylibcudf.libcudf.scalar.scalar_factories cimport (
+    make_string_scalar as cpp_make_string_scalar,
+)
+from cudf._lib.pylibcudf.libcudf.strings.replace cimport (
+    replace as cpp_replace,
+    replace_slice as cpp_replace_slice,
+)
+from cudf._lib.pylibcudf.libcudf.types cimport size_type
+from cudf._lib.pylibcudf.scalar cimport Scalar
+from cudf._lib.pylibcudf.strings.types cimport ColumnOrScalar
+
+
+cpdef Column replace(
+    Column input,
+    ColumnOrScalar target,
+    ColumnOrScalar repl,
+    size_type maxrepl = -1
+):
+    cdef:
+        unique_ptr[column] c_result
+        const string_scalar* target_str
+        const string_scalar* repl_str
+
+    if ColumnOrScalar is Scalar:
+        target_str = <string_scalar *>(target.c_obj.get())
+        repl_str = <string_scalar *>(repl.c_obj.get())
+
+        with nogil:
+            c_result = move(cpp_replace(
+                input.view(),
+                target_str[0],
+                repl_str[0],
+                maxrepl,
+            ))
+    else:
+        # Column case
+        # TODO: maxrepl should be supported in the corresponding CUDA/C++ code
```

Comment:
For the overload of replace in libcudf where input/target/repl are columns, there isn't a maxrepl arg.

We should probably support this in libcudf replace (eventually), otherwise we'll have some weirdness in pylibcudf where we'll have to raise for maxrepl despite accepting it as an argument.

Reply:
Good idea. Can you raise an issue?

In the meantime, I would recommend that we change the default value of the parameter to None, then raise a NotImplementedError in this branch of the code if we find a non-None value, while in the Scalar branch we set it to -1.

Reply:
#15855